### DISCUSSION TITLE
Error: Configuration File Not Found

### DISCUSSION THREAD

**Discussion by mikea213** [qdm12/gluetun#1235] [Q&A] [answered]
Im trying to install vpn into a container and getting the following error and cant figure out what Im doing wrong.
In my folder named gluetun I have all the files I downloaded from my vpn which has the .conf,.ovpn,.key,.crt
I can run openvpn outside the container and it works so I know my files are correct I just want it running in a container so I can assign it to certain ones and the others pull my normal ip. Thanks in advance.

```
2022-11-11T07:54:27Z INFO [routing] default route found: interface eth0, gateway 172.27.0.1 and assigned IP 172.27.0.2
2022-11-11T07:54:27Z INFO [routing] local ethernet link found: eth0
2022-11-11T07:54:27Z INFO [routing] local ipnet found: 172.27.0.0/16
2022-11-11T07:54:28Z INFO [firewall] enabling...
2022-11-11T07:54:28Z INFO [firewall] enabled successfully
2022-11-11T07:54:28Z INFO [storage] creating /gluetun/servers.json with 13220 hardcoded servers
2022-11-11T07:54:29Z ERROR VPN settings: provider settings: server selection: OpenVPN server selection settings: configuration file: file does not exist: gluetun/mikea213-1.conf
2022-11-11T07:54:29Z INFO Shutdown successful
```

Here is what my stack looks like. I did try changing on volumes the config.conf to mine that in the folder to the mikea213-1.conf but then is says that name is a folder and its not.

```yml
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
      - NET_ADMIN
    volumes:
      - /srv/dev-disk-by-uuid-cf5877a4-9fa2-48d2-8f2f-fd5091c2eba2/appdata/gluetun:/gluetun/config.conf:ro
    environment:
      - VPN_SERVICE_PROVIDER=custom
      - VPN_TYPE=openvpn

      # For OpenVPN
      - OPENVPN_CUSTOM_CONFIG=gluetun/mikea213-1.conf
```

**Comment by qdm12** [comment, marked as answer]
- `...appdata/gluetun:/gluetun/config.conf:ro` looks like it bind mounts a directory `gluetun` to a file `config.conf`, I guess what you want is `/srv/dev-disk-by-uuid-cf5877a4-9fa2-48d2-8f2f-fd5091c2eba2/appdata/gluetun:/gluetun`? If you have multiple files, you need to have them all in that gluetun directory as well.
- Check step 3. of https://github.com/qdm12/gluetun/wiki/OpenVPN-configuration-file#setup so use absolute paths for `OPENVPN_CUSTOM_CONFIG`
- Depending what your configuration file is named inside `/srv/dev-disk-by-uuid-cf5877a4-9fa2-48d2-8f2f-fd5091c2eba2/appdata/gluetun` (not sure if it's `config.conf` or `mikea213-1.conf`), update the file name in `OPENVPN_CUSTOM_CONFIG` to match that

**Reply by mikea213** [reply]
Thank you that helped I got it working. Now I just need to figure out how to attach that container to other containers. I have it attached to qbittorrent but for some reason it wont let me connect it to another. Gives me an error saying I can only have it attached to one container at a time.

**Reply by bnhf** [reply]
@mikea213 

You might check the marked answer I gave in this discussion:

https://github.com/qdm12/gluetun/discussions/1172

For an example of how to construct a stack where a given container connects to a Gluetun container. Portainer is a great way to manage the lot, if you're not already using it.
